core/unit_test/TestRange.hpp

```diff
@@ -382,4 +439,47 @@ TEST(TEST_CATEGORY, large_parallel_for_reduce) {
 }
 #endif
 
+TEST(TEST_CATEGORY, check_batch_size) {
+  ASSERT_TRUE(Kokkos::Experimental::StaticBatchSize<1>::batch_size == 1);
+  ASSERT_TRUE(Kokkos::Experimental::StaticBatchSize<4>::batch_size == 4);
+}
+
+TEST(TEST_CATEGORY, range_static_batch_size) {
+  {
+    TestStaticBatchSize<TEST_EXECSPACE,
+                        Kokkos::Experimental::StaticBatchSize<1>>
+        f(1024);
+    f.test_batch_size();
+  }
```

Comment:
This is a suggestion but I'm fine not doing it.

I would propose you pass an execution space instance to avoid all the additional fences `Kokkos` is adding otherwise.

```suggestion
TEST(TEST_CATEGORY, range_static_batch_size) {
  const TEST_EXECSPACE exec{};
  {
    TestStaticBatchSize<TEST_EXECSPACE,
                        Kokkos::Experimental::StaticBatchSize<1>>
        f(exec, 1024);
    f.test_batch_size(exec);
  }
```